### ISSUE TITLE
response_model ignores @property getter

### ISSUE THREAD

**Issue by charlax** [tiangolo/fastapi#2644]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from typing import Optional

from fastapi import FastAPI
from fastapi.testclient import TestClient
from pydantic import BaseModel, EmailStr

app = FastAPI()


class UserIn(BaseModel):
    username: str

    @property
    def email(self):
        return f"{self.username}@foo.com"


class UserOut(BaseModel):
    username: str
    email: str


@app.get("/users/1", response_model=UserOut)
async def get_user():
    return UserIn(username="toaster")



client = TestClient(app)


def test_get():
    response = client.get("/users/1")
    assert response.status_code == 200
    assert response.json() == {"email": "toaster@gdc.com"}
```

### Description

* Run with pytest
* Watch the test fail
* But I expected it to pass

Failure:

```
  File "testfastapi/env/lib/python3.9/site-packages/fastapi/routing.py", line 209, in app
    response_data = await serialize_response(
  File "testfastapi/env/lib/python3.9/site-packages/fastapi/routing.py", line 126, in serialize_response
    raise ValidationError(errors, field.type_)
pydantic.error_wrappers.ValidationError: 1 validation error for UserOut
response -> email
  field required (type=value_error.missing)
```

### Environment

* OS: [e.g. Linux / Windows / macOS]: macOS
* FastAPI Version [e.g. 0.3.0]: 0.63.0
* Python version: 3.9.1

### Additional context

This is probably related to https://github.com/samuelcolvin/pydantic/issues/935 - but I'm wondering if we shouldn't use something like `from_orm` when creating response object, because nothing guarantees (I believe?) that the returned object will be a `BaseModel`?

In any case, thanks a lot for your work on both libraries!

**Comment by Mause** [comment]
The easiest way would be to set orm_mode on UserOut

**Comment by charlax** [comment]
> The easiest way would be to set orm_mode on UserOut

Tried it, but as expected the fastapi code would have to do `from_orm` for this to work as well...

**Comment by Mause** [comment]
Yes, which it will? Did you get another error?

**Comment by charlax** [comment]
Yes, same error...

```python
from typing import Optional

from fastapi import FastAPI
from fastapi.testclient import TestClient
from pydantic import BaseModel, EmailStr

app = FastAPI()


class UserIn(BaseModel):
    username: str

    @property
    def email(self):
        return f"{self.username}@gdc.com"


class UserOut(BaseModel):
    username: str
    email: str

    class Config:
        orm_mode = True


@app.get("/users/1", response_model=UserOut)
async def get_user():
    return UserIn(username="toaster")



client = TestClient(app)


def test_get():
    response = client.get("/users/1")
    assert response.status_code == 200
    assert response.json() == {"email": "toaster@gdc.com"}
```

I could be wrong but you need (1) `orm_mode = True` and (2) `from_orm(...)` (which requires change in `serialize_response`).

**Comment by Mause** [comment]
Nope, it looks like it's not working because fastapi under the hood calls .dict() on the model, which means the later orm_mode functionality won't see the field. You could override this method, or alternatively, do something like this:

```py
class UserIn(BaseModel):
    username: str
    email: Optional[EmailStr] = None
    
    def __init__(self, username, email=None):
        super().__init__(username=username, email=email)
        self.email = f"{self.username}@gdc.com"
```

**Comment by charlax** [comment]
[Marshmallow](https://marshmallow.readthedocs.io/en/stable/) allows using a property... I haven't had time to finish reading the code, but this essentially restricts the kind of objects that can be returned from a router function... Using simple `getattr()` instead of calling `.dict()` would allow for using both Pydantic models AND any other object, including plain objects/ORM objects/`property` on `BaseModel`, a net benefit - what do you think?

**Comment by Mause** [comment]
I mean, this isn't a limitation of FastAPI at all, it's a limitation of pydantic. Ultimately if you wish to use marshmallow instead for your internal models, there's nothing stopping you

**Comment by charlax** [comment]
I'd love to keep everything under the fastapi/pydantic umbrella - I mentioned marshmallow because this shows that it seems an intuitive feature to support...

It's true that some part of this feature should live within pydantic. Do you recommend I open a ticket there too?

**Comment by PrettyWood** [comment]
Hi @charlax!
It is definitely related to _pydantic_! You're welcome to try with https://github.com/samuelcolvin/pydantic/pull/2625 in local to see if this suits your needs :)
